src/table/styling-utils.js

```diff
@@ -1,14 +1,24 @@
 export function getColor(color, defaultColor, theme) {
-  const resolvedColor = theme.getColorPickerColor(color);
-  return !resolvedColor || resolvedColor === 'none' ? defaultColor : resolvedColor;
+  const resolvedColor = color && theme.getColorPickerColor(color);
+  return !resolvedColor || resolvedColor === 'none' ? color || defaultColor : resolvedColor;
```

Comment:
As I commented on the other PR, no need to sanity check `color`, `getColorPickerColor` just returns undefined. Also, this we always want to fall back to `defaultColor` if `resolvedColor` is not valid